### DISCUSSION TITLE
How to customize the bibliography suggestions preview?

### DISCUSSION THREAD

**Discussion by rafisics** [latex-lsp/texlab#1097] [Q&A]
Using Texlab LSP, the bibliography suggestions appear with categories like `Struct`, `Interface`, `Event` etc. 

![Screenshot from 2024-04-27 01-21-34](https://github.com/latex-lsp/texlab/assets/68302816/c10731ab-f094-4016-b8e7-ba04146577db)

But I want to show them as the item types such as a `Book`, `Preprint`, `Journal Article`, etc. I mean, I want to get the preview window as
```
zee2016      - Book
stewart1991  - Book
lee2007      - Preprint
gubser2002   - Preprint
fuchs1995    - Book  
georgi2000   - Book
candelas1990 - Journal Article
```

**Comment by clason** [comment]
This is not something that the LSP protocol really allows, so this is a bit hacky, but here's how I do that (adjust to taste):
https://gist.github.com/clason/3701040203c6625c24eb099cd3ef6d5c#file-lsp-lua-L25-L102

The types are documented here: https://github.com/latex-lsp/texlab/wiki/LSP-Internals#enum-mapping

**Reply by rafisics** [reply]
Thank you. Can you please tell me where I should put that setup in my config? 
https://github.com/rafisics/.config/tree/my-nvim/nvim
